includes/event-date.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Wporg\TranslationEvents;
+
+use DateTime;
+use DateTimeImmutable;
+use DateTimeZone;
+use Exception;
+
+abstract class Event_Date extends DateTimeImmutable {
+	abstract public function is_end_date(): bool;
+	public $timezone;
+	public function __construct( string $date, DateTimeZone $timezone = null ) {
+		if ( ! $timezone ) {
+			$timezone = new DateTimeZone( 'UTC' );
+		}
+
+		$utc_date = new DateTime( $date, new DateTimeZone( 'UTC' ) );
+		$utc_date->setTimezone( $timezone );
+		parent::__construct( $utc_date->format( 'Y-m-d H:i:s' ), $timezone );
+		$this->timezone = $timezone;
+	}
+
+	/**
+	 * Get the standard formatted text for the date in UTC.
+	 *
+	 * @return string The date text.
+	 */
+	public function __toString(): string {
+		return $this->utc( 'Y-m-d H:i:s' );
+	}
+	/**
+	 * Get the local formatted text for the date in UTC.
+	 *
+	 * @return string The date text.
+	 */
+	public function utc( $format ): string {
+		return $this->setTimeZone( new DateTimeZone( 'UTC' ) )->format( $format );
+	}
+
+	/**
+	 * Generate variable text depending on when the event starts or ends.
+	 *
+	 * @return string The end date text.
+	 */
+	public static function get_variable_text(): string {
```

Comment:
This function doesn't seem to be called anywhere. I'm guessing maybe this isn't needed anymore?

Reply:
It is now used